### PR TITLE
Prune unused vendor packages

### PR DESCRIPTION
**Pull request by shashidharatd**
Addresses https://github.com/kubernetes-sigs/federation-v2/issues/413

/cc @marun @kubernetes-sigs/federation-v2-maintainers

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ source = "gopkg.in/fsnotify/fsnotify.v1"
 
 [prune]
   go-tests = true
-  #unused-packages = true
+  unused-packages = true
```
Comment (gyliu513) [inline Gopkg.toml]: Thanks @shashidharatd , one question, how about go through Gopkg.toml and check if we can remove all the unused `[[override]]`?

Comment (shashidharatd) [reply Gopkg.toml]: AFAIK, these override directive will not cause inclusion of unused packages. they are used to override the package version or source etc...

Comment (shashidharatd) [reply Gopkg.toml]: Anyways, i will update the packages based on what is generated by kubebuilde v1.0.4

**Comment by marun** [comment]
I don't think we can prune non-go, as per the ci failure.

**Comment by shashidharatd** [comment]
> I don't think we can prune non-go, as per the ci failure.

i excluded `github.com/kubernetes/repo-infra` from pruning now. should fix the ci issue.

**Comment by shashidharatd** [comment]
CI build is successful now. @marun, @gyliu513 PTAL

**Comment by marun** [comment]
I'm having trouble vendoring k8s.io/kubectl after this merge.  I'm going to see if reverting locally helps.

**Pull request merged**